packages/lucia-auth/src/middleware/index.ts

```diff
@@ -160,3 +161,37 @@ export const web = (): Middleware<[Request, Headers]> => {
 		return requestContext;
 	};
 };
+
+/**
+ * You need to register `setCookie` from:
+ * 
+ * ```
+ * import cookie from '@fastify/cookie';
+ * 
+ * await fastify.register(cookie, {
+ *  secret: 'my-secret', // for cookies signature
+ *  parseOptions: {}, // options for parsing cookies
+ * });
+ * ```
+ */
+// eslint-disable-next-line arrow-body-style
+export const fastify = (): Middleware<[FastifyRequest, FastifyReply]> => {
+	return (request, reply) => {
+		const requestContext = {
+			request: {
+				url: request.url,
+				method: request.method,
+				headers: {
+					origin: request.headers.origin ?? null,
+					cookie: request.headers.cookie ?? null,
+				},
+			},
+			setCookie: (cookie) => {
+				// FIXME: eslint
+				// eslint-disable-next-line @typescript-eslint/no-floating-promises
+				reply.setCookie(cookie.name, cookie.value, cookie.attributes);
```

Comment:
Typing issue

Reply:
`setCookie()` is asynchronous?

Reply:
Whack-a-mole.
Static TS and JS runtime say synchronous is fine, ESLint is whining… maybe a global namespace conflict somewhere?

Reply:
It's also happening to me, must be a bug...